src/main/java/org/humancellatlas/ingest/project/Project.java

```diff
@@ -67,4 +64,18 @@ public SubmissionEnvelope getOpenSubmissionEnvelope(){
         }
         return null;
     }
-}
\ No newline at end of file
+
+    @JsonIgnore
+    public boolean removeSubmissionEnvelope(SubmissionEnvelope submissionEnvelope) {
+        if(!submissionEnvelope.isOpen())
```

Comment:
You could remove the check, just let the caller do it if needed since I think you'll reuse this function in the nocheck delete functionality (for complete submissions).